Add workaround for firebase messaging (#289)
ably/ably-flutter: README.md

See [PushNotifications.md](PushNotifications.md) for detailed information on getting PN working with the example app.

> ###### **Please note:**

Reviewer: I think this needs to be H4, given previous heading is H3... or, in other words, this feels like misuse of a heading so I would perhaps also change the heading text (also makes this deep linkable):

```suggestion
#### Important Note regarding Firebase and Push Notifications
```

> #### [Important Note regarding Firebase and Push Notifications](https://github.com/ably/ably-flutter/issues/226)

Reviewer: This readme actually has two sections with the same "Push Notifications" heading. 🙄 
[This one](https://github.com/ably/ably-flutter/tree/add-workaround_for_firebase_messaging#push-notifications), and [another one later on](https://github.com/ably/ably-flutter/tree/add-workaround_for_firebase_messaging#push-notifications-1).

Perhaps that was the initial impetus for the creation of the [dedicated, discrete document `PushNotifications.md`](https://github.com/ably/ably-flutter/blob/add-workaround_for_firebase_messaging/PushNotifications.md).

Regardless of reasons, the reality is that there are two places that could do with this information being visible... therefore my suggestion would be to move this new content to the dedicated document.

Author: I was aware of that but I thought this would be a bit more prominent if we have this on the original readme. And I also think that the readme here is in needs of shortening, Perhaps we can then move push notification section here. But I will move this to the separate file for now

Author: https://github.com/ably/ably-flutter/pull/289/commits/b72803873bccfe4099a92e0dd9fb999388f288ae

> If you are using [official Firebase messaging library package](https://pub.dev/packages/firebase_messaging) along with this library, you must add following block to your Android application's manifest file under application tag. This is a workaround that prevents a conflict risen from two libraries installed together. 

Reviewer: ```suggestion
If you are using the [official Firebase messaging library package](https://pub.dev/packages/firebase_messaging) along with this library, then you must add the following block to your Android application's manifest file within the `application` element. This is a workaround that prevents a conflict rising from the two libraries installed together. 
```

> If you are using the [official Firebase messaging library package](https://pub.dev/packages/firebase_messaging) along with this library, then you must add the following block to your Android application's manifest file within the `application` element. This is a workaround that prevents a conflict rising from the two libraries installed together. 

Reviewer: ```suggestion
If you are using the [official Firebase messaging library package](https://pub.dev/packages/firebase_messaging) along with this library, then you must add the following block to your Android application's manifest file within the `application` element. This is a workaround that prevents a conflict that happens when the two libraries are installed together.
```

Author: https://github.com/ably/ably-flutter/pull/289/commits/2e6c8ff3cf785104366d69ffccbd5d7e96c432f5
